pydantic_ai_slim/pydantic_ai/models/anthropic.py

```diff
@@ -0,0 +1,399 @@
+            self.client = AsyncAnthropic(api_key=api_key, http_client=cached_async_http_client())
+
+    async def agent_model(
+        self,
+        *,
+        function_tools: list[ToolDefinition],
+        allow_text_result: bool,
+        result_tools: list[ToolDefinition],
+    ) -> AgentModel:
+        check_allow_model_requests()
+        tools = [self._map_tool_definition(r) for r in function_tools]
+        if result_tools:
+            tools += [self._map_tool_definition(r) for r in result_tools]
+        return AnthropicAgentModel(
+            self.client,
+            self.model_name,
+            allow_text_result,
+            tools,
+        )
+
+    def name(self) -> str:
+        return self.model_name
+
+    @staticmethod
+    def _map_tool_definition(f: ToolDefinition) -> ToolParam:
+        return {
+            'name': f.name,
+            'description': f.description,
+            'input_schema': f.parameters_json_schema,
+        }
+
+
+@dataclass
+class AnthropicAgentModel(AgentModel):
+    """Implementation of `AgentModel` for Anthropic models."""
+
+    client: AsyncAnthropic
+    model_name: str
+    allow_text_result: bool
+    tools: list[ToolParam]
+
+    async def request(self, messages: list[Message]) -> tuple[ModelAnyResponse, result.Cost]:
+        response = await self._messages_create(messages, False)
+        return self._process_response(response), _map_cost(response)
+
+    @asynccontextmanager
+    async def request_stream(self, messages: list[Message]) -> AsyncIterator[EitherStreamedResponse]:
+        response = await self._messages_create(messages, True)
+        async with response:
+            yield await self._process_streamed_response(response)
+
+    @overload
+    async def _messages_create(
+        self, messages: list[Message], stream: Literal[True]
+    ) -> AsyncStream[RawMessageStreamEvent]:
+        pass
+
+    @overload
+    async def _messages_create(self, messages: list[Message], stream: Literal[False]) -> AnthropicMessage:
+        pass
+
+    async def _messages_create(
+        self, messages: list[Message], stream: bool
+    ) -> AnthropicMessage | AsyncStream[RawMessageStreamEvent]:
+        # standalone function to make it easier to override
+        if not self.tools:
+            tool_choice: ToolChoiceParam | None = None
+        elif not self.allow_text_result:
+            tool_choice = {'type': 'any'}
+        else:
+            tool_choice = {'type': 'auto'}
+
+        anthropic_messages = [self._map_message(m) for m in messages]
+        return await self.client.messages.create(
+            # TODO: might want to change max tokens (or make configurable for user), and same with other models...
+            max_tokens=1024,
+            messages=anthropic_messages,
+            model=self.model_name,
+            temperature=0.0,
+            tools=self.tools or NOT_GIVEN,
+            tool_choice=tool_choice or NOT_GIVEN,
+            stream=stream,
+        )
```

Comment:
Curious about this one - what's our way forward here? I feel like more advanced users will want more control over these types of things?

Reply:
I guess this is something to be addressed in a different PR?

Reply:
We discussed the fact that we'll add a `TypedDict` to run with common settings in a different PR.